pyanaconda/modules/common/submodule_manager.py

```diff
@@ -0,0 +1,63 @@
+#
+# Submodule manager for kickstart services.
+#
+# Copyright (C) 2023 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from collections import OrderedDict
+
+__all__ = ["SubmoduleManager"]
+
+class SubmoduleManager:
+    """Kickstart sub-module manager.
+
+    This class helps manage multiple kickstart service sub-modules.
+    """
+
+    def __init__(self):
+        self._modules = OrderedDict()
+
+    def add_module(self, name, module):
+        """Add a module."""
+        self._modules[name] = module
+
+    def __getitem__(self, item):
```

Comment:
It would be great if you could also iterate over the list of modules.

Reply:
Yes, I considered that too. Ultimately, I think that's something to add only when needed. After all, `__getitem__` is there only to support the getting, I would just dump it otherwise. Also, having access to the underlying list opens the can of worms that is direct access to the "managed" list. I would prefer keeping this as simple as possible, until needed.

In other words, if there are some places where having an iterable like that makes sense, feel free to share, that's a data-based reason to have it.

Reply:
I now realize the other comment above might be related to this. IMO the point still stands - let's add this once needed.